### ISSUE TITLE
Some other federated instances are attempting to POST to the wrong URL (and failing)

### ISSUE THREAD

**Issue by SteveDinn** [pixelfed/pixelfed#4410]
I had noticed that I was never receiving any posts from some of the users that I was following via Pixelfed, despite them posting photos regularly. I was stumped by this, until I happened to notice in my web logs that two instances were repeatedly attempting to POST to an invalid URL and getting 404 errors.

`//f/inbox` instead of `/f/inbox`. It just so happens that these are the two instances hosting the accounts from which I was not receiving posts.

Both *hachyderm.io* and *mastodon.art* seem to have this incorrect URL for my instance. They are both running Mastodon 4.1.2, but I don't think that has anything to do with it, because other instances running the same software are POSTing to the correct URL. Is there any way I can correct this issue with the other instances?

**Comment by trwnh** [comment]
did you add a trailing slash when setting up your pixelfed domain in the config file?

**Comment by SteveDinn** [comment]
There is not a trailing slash there now. I might have had one there when my instance was just a baby and I was still figuring things out, but I'm not sure about that.

Is there any way to correct what those other instances think the inbox URL for my instance is? 

Edit: it seems that those `//f/inbox` URLs are the only ones with an extra `/`. Even more, hachyderm.io seems to have corrected itself earlier today. I don't know what would have changed on my end  At some point, it just started using the correct URL:
```
88.99.61.61 - [2023-05-22 16:56:28.070] "pixelfed.dinn.ca" POST //f/inbox HTTP/1.1 404 4997 "" "http.rb/5.1.1 (Mastodon/4.1.2; +https://hachyderm.io/)"
88.99.61.61 - [2023-05-22 19:31:51.153] "pixelfed.dinn.ca" POST //f/inbox HTTP/1.1 404 4997 "" "http.rb/5.1.1 (Mastodon/4.1.2; +https://hachyderm.io/)"
195.201.195.102 - [2023-05-23 01:49:44.568] "pixelfed.dinn.ca" POST //f/inbox HTTP/1.1 404 4997 "" "http.rb/5.1.1 (Mastodon/4.1.2; +https://hachyderm.io/)"
88.99.61.61 - [2023-05-23 03:53:03.664] "pixelfed.dinn.ca" POST /f/inbox HTTP/1.1 200 25 "" "http.rb/5.1.1 (Mastodon/4.1.2; +https://hachyderm.io/)"
88.99.61.61 - [2023-05-23 06:40:37.719] "pixelfed.dinn.ca" POST /f/inbox HTTP/1.1 200 25 "" "http.rb/5.1.1 (Mastodon/4.1.2; +https://hachyderm.io/)"
88.99.61.61 - [2023-05-23 17:47:47.825] "pixelfed.dinn.ca" POST /f/inbox HTTP/1.1 200 25 "" "http.rb/5.1.1 (Mastodon/4.1.2; +https://hachyderm.io/)"
```

Mastodon.art is still messed up for me though:
```
57.128.95.173 - [2023-05-23 15:33:54.247] "pixelfed.dinn.ca" POST //f/inbox HTTP/1.1 404 4997 "" "http.rb/5.1.1 (Mastodon/4.1.2; +https://mastodon.art/)"
57.128.95.173 - [2023-05-23 15:34:06.621] "pixelfed.dinn.ca" POST //f/inbox HTTP/1.1 404 4997 "" "http.rb/5.1.1 (Mastodon/4.1.2; +https://mastodon.art/)"
57.128.95.173 - [2023-05-23 15:45:01.193] "pixelfed.dinn.ca" POST //f/inbox HTTP/1.1 404 4997 "" "http.rb/5.1.1 (Mastodon/4.1.2; +https://mastodon.art/)"
57.128.95.173 - [2023-05-23 16:09:05.325] "pixelfed.dinn.ca" POST //f/inbox HTTP/1.1 404 4997 "" "http.rb/5.1.1 (Mastodon/4.1.2; +https://mastodon.art/)"
57.128.95.173 - [2023-05-23 18:40:42.112] "pixelfed.dinn.ca" POST //f/inbox HTTP/1.1 404 4997 "" "http.rb/5.1.1 (Mastodon/4.1.2; +https://mastodon.art/)"
57.128.95.173 - [2023-05-23 18:58:30.815] "pixelfed.dinn.ca" POST //f/inbox HTTP/1.1 404 4997 "" "http.rb/5.1.1 (Mastodon/4.1.2; +https://mastodon.art/)"
57.128.95.173 - [2023-05-23 20:47:28.032] "pixelfed.dinn.ca" POST //f/inbox HTTP/1.1 404 4997 "" "http.rb/5.1.1 (Mastodon/4.1.2; +https://mastodon.art/)"
57.128.95.173 - [2023-05-23 22:26:36.527] "pixelfed.dinn.ca" POST //f/inbox HTTP/1.1 404 4997 "" "http.rb/5.1.1 (Mastodon/4.1.2; +https://mastodon.art/)"
```

**Comment by trwnh** [comment]
possibly they fetched an older version of your profile and then fetched an updated version of your profile, then? i think you *might* be able to force send an Update activity to mastodon.art's sharedInbox to get it to do the same, but there isn't a built-in way to do this in Pixelfed. you can only have someone on mastodon.art force-refresh your profile i guess.

**Comment by SteveDinn** [comment]
I was able to contact the admin of mastodon.art and get them to refresh whatever needed to be refreshed and that seems to have set things right, at least as far as the incoming POST web requests in my logs.

I don't know if this is an issue with Pixelfed or not, so I'll leave it open and someone more knowledgeable than me can close it.

**Comment by trwnh** [comment]
it's not a "bug", more a wrong config, but it could certainly have better handling. pixelfed should handle trailing slashes input by admins, and remove them. it could also be a good idea to add a function in the cli and admin dashboard to force-send an Update for a profile to a specific inbox or sharedInbox.

**Comment by SteveDinn** [comment]
Yeah, there's [a handy class in dotNET](https://learn.microsoft.com/en-us/dotnet/api/system.io.path.combine?view=net-7.0) that lets you put path parts together regardless of whether one or both of them have slashes on their adjoining ends. I wonder if there is something analogous in PHP.
